src/main/java/com/siit/hospital_manager/controller/AppointmentController.java

```diff
@@ -45,15 +47,20 @@ public String create(Model model) {
     }
 
     @PostMapping("/submitCreateAppointmentForm")
-    public String submitCreateAppointmentForm(@Valid CreateAppointmentDto createAppointmentDto, Principal principal){
+    public String submitCreateAppointmentForm(@Valid CreateAppointmentDto createAppointmentDto, Principal principal) {
         appointmentService.createAppointment(createAppointmentDto, principal.getName());
         return "redirect:/appointment/findAllByPatient";
     }
 
     @PutMapping("/{id}")
     @ResponseStatus(HttpStatus.ACCEPTED)
-    public void updateAppointment(@PathVariable Integer id, @Valid @RequestBody UpdateAppointmentDto updateAppointmentDto, Principal principal){
+    public void updateAppointment(@PathVariable Integer id, @Valid @RequestBody UpdateAppointmentDto updateAppointmentDto, Principal principal) {
         appointmentService.updateAppointment(id, updateAppointmentDto, principal.getName());
     }
 
+    @GetMapping("/history")
+    public String findHistoryByPatient(Model model, Principal principal) {
+        model.addAttribute("appointments", appointmentService.findPastAppointmentsByUserName(principal.getName()));
```

Comment:
usually we use a date filter instead of separate endpoints for future and past entities